File: src/api/reflog.ts
import { invoke } from "@tauri-apps/api/core";
import type { ReflogEntry } from "@/types/reflog";

/** Read a reflog (default HEAD), newest first. */
export function getReflog(
  repoPath: string,
  reference?: string,
  max?: number,
): Promise<ReflogEntry[]> {
  return invoke<ReflogEntry[]>("get_reflog", {
    repoPath,
    reference: reference ?? null,
    max: max ?? null,
  });
}

File: src-tauri/src/commands/reflog.rs
//! Reflog command (T-14). Recovery ops reuse T-09 `create_branch` and T-13
//! `reset_to` commands directly; this is the read side only.

use std::path::Path;

use crate::core::reflog::{self, ReflogEntry};
use crate::error::AppResult;

/// Read a reflog (default HEAD), newest first. `reference` accepts "HEAD",
/// a full ref ("refs/heads/main"), or a shorthand ("main" / "origin/main").
#[tauri::command]
pub fn get_reflog(
    repo_path: String,
    reference: Option<String>,
    max: Option<usize>,
) -> AppResult<Vec<ReflogEntry>> {
    reflog::read_reflog(
        Path::new(&repo_path),
        reference.as_deref(),
        max.unwrap_or(200),
    )
}
